### ISSUE TITLE
Autosize datatable columns

### ISSUE THREAD

**Issue by MerlinDuChaos** [primefaces/primeng#1009]
Hi,

I think this would be a great addition to the DataTable component to be able to specify that a column should resize either to its content's width or header width.

Thanks !

**Comment by hhubik** [comment]
Yes, please. I want this too.  When I double-click a column header side border, I expect the column to re-size itself to fit the content (like in Excel).

**Comment by heavydawson** [comment]
Great suggestion

**Comment by cagataycivici** [comment]
```
[tableStyle]="{'table-layout':'auto'}"
```

or via style class;

```
.ui-datatable table {
    table-layout: auto
}
```

Note that auto layout is not supported for resize or scrollable tables.

**Comment by cvtoro** [comment]
is there any way to get to enforce 'table-layout: auto' on scrollable tables?

**Comment by Olddude** [comment]
You can achieve the same `table-layout: auto` behavior on scrollable tables when you use the following configuration.

1.  set the `<p-datatable [tableStyle]="{'table-layout':'auto'}" [scrollable]="false">`
2. set the inside the p-datatable css `.ui-datatable-tablewrapper { overflow-x: auto;  }`

**Comment by borisbrodski** [comment]
@Olddude This is an interesting idea, but the table you will get, will scroll (vertically) together with the header. Any ideas, how to achieve the proper scrolling (so that the header of the table always stays above the table body) and still have the nice "table-layout: auto"-like behavior?
Thank you!

**Comment by nasirpinnacle** [comment]
Greatest idea, easy workability:
https://sites.google.com/site/christophenoel/home/primefacesdatatabledynamiccolumnswidth

i coded as

<p:columns value="#{liveRangeService.tableHeaderNames}"
                               var="mycolHeader" width="#{colIndex==0?'10':
                                                          colIndex==1?'150':
                                                          colIndex==2?'10':
                                                          colIndex==3?'10':
                                                          '0'}" columnIndexVar="colIndex" selectRow="true">

**Comment by rudi1602** [comment]
fixed for PrimeFaces by primefaces/primefaces#14387
